app/models/recipe.rb

```diff
@@ -10,4 +10,6 @@ class Recipe < ApplicationRecord
   has_one_attached :picture do |attachable|
     attachable.variant :thumb, resize_to_limit: [100, 100]
   end
+
+  scope :get_recipe_which_has_ingredient, ->(ingredient) { where("recipes.ingredients->'needed' @> ?", [{ name: ingredient }].to_json) }
```

Comment:
It's kinda redundant having 'recipe' in the scope name.
I'd rather go with `scope :with_ingredient`. So, when you consume the scope would look like:
```ruby
Recipe.with_ingredient(item)
```

Reply:
Agree on that 👆🏻
We're already talking about recipes so...